lib/util.js

```diff
@@ -504,49 +505,60 @@ const isArray = exports.isArray = Array.isArray;
 function isBoolean(arg) {
   return typeof arg === 'boolean';
 }
-exports.isBoolean = isBoolean;
+exports.isBoolean = deprecate(isBoolean
+  , 'util.is* functions are deprecated, please use a user-land alternative.');
 
 function isNull(arg) {
   return arg === null;
 }
-exports.isNull = isNull;
+exports.isNull = deprecate(isNull
+  , 'util.is* functions are deprecated, please use a user-land alternative.');
 
 function isNullOrUndefined(arg) {
   return arg === null || arg === undefined;
 }
-exports.isNullOrUndefined = isNullOrUndefined;
+exports.isNullOrUndefined = deprecate(isNullOrUndefined
+  , 'util.is* functions are deprecated, please use a user-land alternative.');
 
 function isNumber(arg) {
   return typeof arg === 'number';
 }
-exports.isNumber = isNumber;
+exports.isNumber = deprecate(isNumber
+  , 'util.is* functions are deprecated, please use a user-land alternative.');
 
 function isString(arg) {
   return typeof arg === 'string';
 }
-exports.isString = isString;
+exports.isString = deprecate(isString
+  , 'util.is* functions are deprecated, please use a user-land alternative.');
 
 function isSymbol(arg) {
   return typeof arg === 'symbol';
 }
-exports.isSymbol = isSymbol;
+exports.isSymbol = deprecate(isSymbol
+  , 'util.is* functions are deprecated, please use a user-land alternative.');
 
 function isUndefined(arg) {
   return arg === undefined;
 }
-exports.isUndefined = isUndefined;
+exports.isUndefined = deprecate(isUndefined
+  , 'util.is* functions are deprecated, please use a user-land alternative.');
 
+// note: the isRegExp function is still used here in util
 function isRegExp(re) {
   return re !== null && typeof re === 'object' &&
          objectToString(re) === '[object RegExp]';
```

Comment:
Wouldn't the `objectToString` be sufficient here? Why do we need the other two checks?